crates/bevy_math/src/curve/easing.rs

```diff
@@ -0,0 +1,328 @@
+//! Module containing different [`Easing`] curves to control the transition between two values and
+//! the [`EasingCurve`] struct to make use of them.
+
+use crate::{
+    ops::{self, FloatPow},
+    prelude::CubicSegment,
+    VectorSpace,
+};
+
+use super::{Curve, FunctionCurve, Interval};
+
+/// A trait for [`Curves`] that map the [unit interval] to some other values. These kinds of curves
+/// are used to create a transition between two values. Easing curves are most commonly known from
+/// [CSS animations] but are also widely used in other fields.
+///
+/// [unit interval]: `Interval::UNIT`
+/// [`Curves`]: `Curve`
+/// [CSS animations]: https://developer.mozilla.org/en-US/docs/Web/CSS/easing-function
+pub trait Easing<T>: Curve<T> {}
+impl<T: VectorSpace, C: Curve<f32>> Easing<T> for EasingCurve<T, C> {}
+impl<T: VectorSpace> Easing<T> for LinearCurve<T> {}
+impl Easing<f32> for CubicBezierCurve {}
+impl Easing<f32> for StepCurve {}
+impl Easing<f32> for ElasticCurve {}
+
+/// A [`Curve`] that is defined by a `start` and an `end` point, together with an [`EasingCurve`]
```

Comment:
The fact that `start` and `end` are also used in the context of `Interval` makes this a little bit confusing. Maybe some explicit clarification on that would be good?